File: backend/app/domain/entities/athlete.py
"""Athlete entity - synced from Arena API"""
from datetime import datetime
from typing import Optional
from uuid import UUID
from sqlmodel import Field, SQLModel, UniqueConstraint


class AthleteBase(SQLModel):
    """Base Athlete fields shared across schemas"""
    uid: UUID = Field(index=True)
    team_id: Optional[int] = None
    sport_event_id: Optional[int] = None
    weight_category_id: Optional[int] = None
    is_competing: Optional[bool] = None
    person_id: Optional[int] = None


class Athlete(AthleteBase, table=True):
    """Athlete model - synced from Arena API"""
    __tablename__ = "athletes"
    __table_args__ = (
        UniqueConstraint("sport_event_id", "person_id", "weight_category_id", name="uq_athlete_event_person_wc"),
    )

    id: Optional[int] = Field(default=None, primary_key=True)
    uid: UUID = Field(index=True)
    person_id: Optional[int] = Field(default=None, foreign_key="persons.id")
    sync_timestamp: datetime = Field(default_factory=datetime.utcnow)

File: frontend/src/types/athlete.ts
// Shared type definitions for Athlete entity

export interface Athlete {
  id: number
  person_full_name: string
  team_id?: number | null
  sport_event_id?: number
  weight_category_id?: number | null
  is_competing?: boolean
  country_iso_code?: string
  weight_category?: string
  sync_timestamp?: string
}
export interface AthleteWithDetails extends Athlete {
  team_name?: string
  weight_category_name?: string
  country_name?: string
}

export interface AthletesResponse {
  athletes: Athlete[]
  count?: number
  success?: boolean
}
